### PR TITLE
Closes #7611: Add UI tests for content context menus

### LINKED ISSUE #7611: [Bug] Add UI tests for content context menus
Support https://testrail.stage.mozaws.net/index.php?/suites/view/3192&group_by=cases:section_id&group_order=asc&group_id=31660 with automated tests

Covering

  |   | ID | Title |   |   |  
-- | -- | -- | -- | -- | -- | --
  |   | C243837 | Verify Long-click "Open link in new tab" |   |   |  
  |   | C244655 | Verify Long-click "Open link in new private tab" |   |   |  
  |   | C243832 | Verify Long-click "Copy link" |   |   |  
  |   | C243838 | Verify Long-click "Share link" |   |   |  
  |   | C243833 | Verify Long-click "Open Image in New Tab" |   |   |  
  |   | C243835 | Verify Long-click "Save image" |   |   |  
  |   | C243834 | Verify Long-click "Copy image location" |   |   |  
  |   | C352050 | Verify Long-click Link Menu Selection List Variations

### PR DESCRIPTION
**Pull request by AaronMT**
See #7611

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("Link 1")
+            verifyLinkContextMenuItems(genericURL.url)
+            clickContextShareLink()
+            // verify share intent is matched with associated type
+            intended(
```
Comment (KMaragh) [inline app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt]: Can this be moved into a method within the browser robot or maybe the test helper? This way the readability of the test stays intact.

File: app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            clickContextSaveImage()
+        }
+
+        downloadRobot {
+        }.clickAllowPermission {
+            verifyDownloadNotificationPopup()
+        }.clickOpen {
+            // verify open intent is matched with associated type
+            intended(
+                allOf(
```
Comment (KMaragh) [inline app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt]: Same as above, move into the appropriate robot.

File: app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt (hunk 1 above)
Comment (AaronMT) [reply app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt]: Moved into the associated robot.

File: app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt (hunk 2 above)
Comment (AaronMT) [reply app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt]: Moved into the associated robot.

**Pull request merged**